src/m365/spo/commands/group/group-member-add.ts

```diff
@@ -171,11 +178,11 @@ class SpoGroupMemberAddCommand extends SpoCommand {
 
     const validUserNames: string[] = [];
     const invalidUserNames: string[] = [];
-    const userInfo: string = args.options.userName ? args.options.userName : args.options.email!;
+    const userInfo: string = args.options.userName ? args.options.userName : args.options.email ? args.options.email : args.options.userId!.toString();
 
     return Promise
-      .all(userInfo.split(',').map(singleUserName => {
-        const options: AadUserGetCommandOptions = {
+      .all(userInfo.split(',').map(async singleUserName => {
+        const options: AadUserGetCommandOptions | SpoUserGetCommandOptions = {
```

Comment:
Hi Nico, I find it hard to understand this section of code. That's partly because the two things (SpoUserGet and AadUserGet) mingle and because it's just too large. Could you refactor it into multiple functions that each do their own thing? That way it will probably become a bit clearer what's going on. Thanks in advance!

Maybe it would be easiest to refactor the entire promise chain into clear async functions.